src/components/webView/WebViewScreen.js

```diff
@@ -12,12 +13,24 @@ export const createWebViewScreen = (source, title) => {
   const CurrentWebView = props => {
     const { navigation } = props
     let loginToken
+    let queryParams
+    let splittedSource
+    let sourceDomain
+    let sourceQuery
 
     switch (title) {
       case 'Rewards':
         loginToken = _get(navigation, 'state.params.loginToken')
+        splittedSource = source.split('?')
+        sourceDomain = splittedSource[0]
+        sourceQuery = splittedSource[1] || ''
+        queryParams = qs.parse(sourceQuery)
 
-        source += `?token=${loginToken}`
+        if (!queryParams.token || queryParams.token !== loginToken) {
+          queryParams.token = loginToken
+        }
+
+        source = `${sourceDomain}?${qs.stringify(queryParams)}`
```

Comment:
Why do we need all of this changes? what was wrong with the previous code?

Reply:
@sirpy the reason for these changes is that every time user clicks on rewards button login token was added to source string again and again, and the string growing like that https://w3.gooddollar.org?token=${token}?token=${https://w3.gooddollar.org/}?token=${https://w3.gooddollar.org/}

Reply:
The link on the rewards tabs should be constant meaning it is always config.w3Url+?loginToken=X
dont pass params to the webview.

Reply:
@sirpy the best way I found to solve this task is to wait for login token to be async fetched from Usertorage and then pass it to createWebViewScreen function where the source string will be built and then passed to WebView Component. As the login token can be fetched from UserStorage asynchronously - that was not easy to do that.

Reply:
@yaroslav-fedyshyn-nordwhale looking also at the iframe issue
i've created #613 
it uses a custom component for rewards that async wait for the token.
it also solves the responsive issue, but introduces another scrolling issue